types/src/shared/utils/string_utils.ts

```diff
@@ -75,3 +75,11 @@ export function truncate(text: string, length: number, omission = "...") {
     ? `${text.substring(0, length - omission.length)}${omission}`
     : text;
 }
+
+export function safeParseJSON(str: string): object | null {
```

Comment:
I would change that function a little for the following reasons:
- returning null => it's an issue because "null" is also valid JSON, callers won't distinguish; we can return undefined or a specific other value;
- catching all => here we just want to catch SyntaxError if I understand your intent (IMO catch alls are bad in general, happy to discuss and explain why if you disagree)
Is it ok for you to address that?

Reply:
It's perfectly fine to just wrap `JSON.parse` without a detailed catch, as long as it only involves JSON.parse. In other cases, I agree that being specific about the error type is the way to go. Instead of using random values like null or undefined, I'll opt for a `Result` to handle it more cleanly.

Reply:
There's a subject for debate here, happy to do that on 80/20 when you're back :)
But your changes are :+1: , approving